tests/smoke/test_default_import_boundary.py

```
@@ -0,0 +1,40 @@
+import cli.dedupe_hash       # must not raise (hash-based dedup path)
+from services.deduplication.detector import DuplicateDetector  # must not raise
+import services.search        # must not raise (HybridRetriever guard required)
+"""
+    env = os.environ.copy()
+    # Prepend src/ so imports work from a clean checkout before `pip install -e .`
+    # On CI the editable install covers this, but local runs may not have it.
+    # Use os.pathsep (';' on Windows, ':' on POSIX) for cross-platform portability.
+    src_path = str(Path(__file__).resolve().parents[2] / "src")
+    existing = env.get("PYTHONPATH", "")
+    env["PYTHONPATH"] = src_path + (os.pathsep + existing if existing else "")
+    result = subprocess.run(
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)